### ISSUE TITLE
Length information in error message of sanitize_index

### ISSUE THREAD

**Issue by vishal-srini-rao** [pandas-dev/pandas#26967]
It may be helpful for quicker debugging to provide information of data and index lengths in the error message in sanitize_index function:
https://github.com/pandas-dev/pandas/blob/2115bf3d58ecc20aa2b10e05574eaa06de1e56ec/pandas/core/internals/construction.py#L516

The modified code could be:
raise ValueError(f'Length of values {len(data)} does not match length of index {len(index)}')

**Comment by WillAyd** [comment]
Thanks for the suggestion, but I personally don't see that being much more incrementally useful. 

Let's see if anyone else sees value otherwise will close out in a few days

**Comment by TomAugspurger** [comment]
This seems reasonable to me. A PR updating the error message and a test would be welcome.

**Comment by WillAyd** [comment]
Is this solved by #26911?

**Comment by TomAugspurger** [comment]
Not sure if it was #26911, but we should be OK now.

```
In [1]: import pandas as pd
 =
In [2]: pd.Series([0,1], index=[0, 1, 2])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-23ab3324fbf7> in <module>
----> 1 pd.Series([0,1], index=[0, 1, 2])

~/sandbox/pandas/pandas/core/series.py in __init__(self, data, index, dtype, name, copy, fastpath)
    241                             'Length of passed values is {val}, '
    242                             'index implies {ind}'
--> 243                             .format(val=len(data), ind=len(index)))
    244                 except TypeError:
    245                     pass

ValueError: Length of passed values is 2, index implies 3
```

LMK if you were taking a different path @vishal-srini-rao
